src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregatingTagger.cs

```diff
@@ -174,10 +174,6 @@ private void OnDiagnosticsUpdated(object sender, DiagnosticsUpdatedArgs e)
 
             private void OnDiagnosticsUpdated(DiagnosticsUpdatedArgs e)
             {
-                // First see if this is a document/project removal.  If so, clear out any state we
-                // have associated with any analyzers we have for that document/project.
-                ProcessRemovedDiagnostics(e);
```

Comment:
Specifically, when a document is opened in the editor, the MiscellaneousWorkspace first takes ownership, and later the RoslynVS workspace takes ownership and MiscellaneousWorkspace detaches from the document. I am concerned that we might retain stale squiggles, if the diagnostics were computed and squiggles added while MiscellaneousWorkspace still had document ownership, and then RoslynVS workspace takes document ownership, and then diagnostics removed for document detach from MiscellaneousWorkspace (which we now ignore as document has changed workspace), and then we add diagnostics/squiggles for document in Roslyn workspace.